MetaMask/metamask-extension
# Direct users to testnet toggle on dapp requests

This PR is to auto switch the toggle if dapp requests to switch to testnet and the user approves it

* Fixes [#658](https://github.com/MetaMask/MetaMask-planning/issues/658)

## Screencaps

### Before


https://github.com/MetaMask/metamask-extension/assets/39872794/1d66d032-8647-4be2-b2dd-82b836a7165d


### After

https://github.com/MetaMask/metamask-extension/assets/39872794/a766c510-6889-43da-a8b3-6e7851bdec0a



## Manual Testing Steps

- Check testnet toggle is off
- From a dapp make a request to switch to test net
- Approve the request in MetaMask
- Go to network list, check the updates network is test network and toggle to show test network is on


## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.
